ui/js/components/auth.js

```diff
@@ -0,0 +1,165 @@
+             * for the tc-login, which will get closed if it's successful.
+             */
+            ctrl.login = function () {
+                var hash = encodeURIComponent("#");
+                var colon = encodeURIComponent(":");
+                var target = `${$location.protocol()}${colon}//${$location.host()}${colon}${$location.port()}/${hash}/login`;
+                var description = "Treeherder";
+                var url = `https://login.taskcluster.net/?target=${target}&description=${description}`;
+
+                // open a new tab to show the taskcluster auth login page
+                $window.open(url, "_blank");
+            };
+
+            /**
+             * Contact Treeherder back-end to log the user out and invalidate
+             * the session token.  Then updates the UI
+             */
+            ctrl.logout = function () {
+                $http.get(thUrl.getRootUrl("/auth/logout/"));
+                ctrl.setLoggedOut();
+            };
+
+            ctrl.setLoggedIn = function(newUser) {
+                newUser.loggedin = true;
+                localStorageService.set("user", newUser);
+                _.extend(ctrl.user, newUser);
+                ctrl.onUserChange({$event: {user: ctrl.user}});
+
+            };
+
+            ctrl.setLoggedOut = function() {
+                localStorageService.set("user", loggedOutUser);
+                _.extend(ctrl.user, loggedOutUser);
+                ctrl.onUserChange({$event: {user: loggedOutUser}});
+            };
+        }]
+});
+
+treeherder.component("loginCallback", {
+    template: `
+        <div ng-if="!loginError">
+            <span>Logging in... <span class="fa fa-spinner fa-spin"></span></span>
+        </div>
+        <div ng-if="loginError">
+            <span>Error logging in: {{loginError}}</span>
+        </div>
+    `,
+    controller: ['localStorageService', '$location', '$window', '$http', '$scope',
+        function(localStorageService, $location, $window, $http, $scope) {
+            const host = $location.host();
+            const port = $location.port();
+            const urlBase = `${$location.protocol()}://${host}:${port}/`;
+            const certificate = $location.search().certificate;
+            const credentials = {
+                id: $location.search().clientId,
+                key: $location.search().accessToken,
+                algorithm: 'sha256'
+            };
+            this.loginError = null;
+            const header = hawk.client.header(urlBase, 'GET', {
+                credentials: credentials,
+                ext: hawk.utils.base64urlEncode(JSON.stringify({"certificate": JSON.parse(certificate)}))}
+            );
+
+            // send a request from client side to TH server signed with TC
+            // creds from login.taskcluster.net
+            $http.get(`${urlBase}api/auth/login/`,
+                      {headers: {"oth": header.field}})
```

Comment:
Should we be POSTing the auth header to this endpoint instead? (Same for logout) Something doesn't seem right about making this a GET request, since it's not nullipotent. 

I'd also then prefer to either put the auth header in the body of the POST request, or else come up with another header name then "OTH". I'd say "Authorization" but I wonder if our own Hawk API middleware will grab it instead? (http://stackoverflow.com/questions/7802116/custom-http-authorization-header may also be of use)